### PR TITLE
build: remove endorsments

### PR DESCRIPTION
**Pull request by Schlagonia**

### CODE REVIEW

File: contracts/registry/Registry.sol [hunk 1]
```diff
@@ -469,4 +411,135 @@ contract Registry is Governance {
...
+
+    /**
+     * @notice Removes a specific `_asset` at `_index` from `assets`.
+     * @dev Can be used if an asset is no longer in use after a vault or
+     * strategy has also been removed.
+     *
+     * @param _asset The asset to remove from the array.
+     * @param _index The index it sits at.
+     */
+    function removeAsset(
```
Comment (wavey0x) [inline contracts/registry/Registry.sol:519]: currently, governance has to remember to call this.
is there a good way to ensure this gets called automatically (when appropriate) when/if all strats/vaults get removed for this asset?

**Review by wavey0x — APPROVED** [review]

File: contracts/registry/Registry.sol (hunk 1 above)
Comment (Schlagonia) [reply contracts/registry/Registry.sol:519]: Yes. Thats how i had made it orginally.
https://github.com/yearn/vault-periphery/blob/5f48dad2ae2c5e528827a2aefd189d43b598250f/contracts/registry/Registry.sol#L438-L447

https://github.com/yearn/vault-periphery/blob/5f48dad2ae2c5e528827a2aefd189d43b598250f/contracts/registry/Registry.sol#L523-L539

Problem is that auto removing it has to iterate over the entire assets array. Which could possibly be 30-40 + items.

So it seemed better to not force the gas cost of that and allow governance to just remove it manually if needed like the strategies and vaults.

I dont think it should cause any real problems for it to be left in even if not in use.

File: contracts/registry/Registry.sol [hunk 2]
```diff
@@ -469,4 +411,135 @@ contract Registry is Governance {
...
+        _strategies[_index] = _strategies[_strategies.length - 1];
+
+        // Set to storage.
+        _endorsedStrategies[asset] = _strategies;
+
+        // Pop the last item off the array.
+        _endorsedStrategies[asset].pop();
+
+        // Reset the info config.
+        info[_strategy] = Info({
```
[the remark this answers is not in the document]
Comment (Schlagonia) [reply contracts/registry/Registry.sol]: https://github.com/yearn/vault-periphery/pull/16/commits/a151f1e0be8bc74812ff3a85b862f67023b0f02e

File: contracts/registry/Registry.sol [hunk 3]
```diff
@@ -469,4 +411,135 @@ contract Registry is Governance {
...
+        _vaults[_index] = _vaults[_vaults.length - 1];
+
+        // Set to storage.
+        _endorsedVaults[asset] = _vaults;
+
+        // Pop the last item off the array.
+        _endorsedVaults[asset].pop();
+
+        // Reset the info config.
+        info[_vault] = Info({
```
[the remark this answers is not in the document]
Comment (Schlagonia) [reply contracts/registry/Registry.sol]: https://github.com/yearn/vault-periphery/pull/16/commits/a151f1e0be8bc74812ff3a85b862f67023b0f02e

File: contracts/registry/Registry.sol [hunk 4]
```diff
@@ -469,4 +411,135 @@ contract Registry is Governance {
...
+            asset: address(0),
+            releaseVersion: 0,
+            deploymentTimeStamp: 0,
```
[the remark this answers is not in the document]
Comment (Schlagonia) [reply contracts/registry/Registry.sol]: https://github.com/yearn/vault-periphery/pull/16/commits/a151f1e0be8bc74812ff3a85b862f67023b0f02e

**Pull request merged**